arch/toolchains/ecmwf-hpc2020-nvhpc.cmake

```diff
@@ -16,15 +16,19 @@ set( ECBUILD_FIND_MPI ON )
 # OpenMP FLAGS
 ####################################################################
 
-set( OpenMP_C_FLAGS             "-mp -mp=bind,allcores,numa" )
-set( OpenMP_CXX_FLAGS           "-mp -mp=bind,allcores,numa" )
-set( OpenMP_Fortran_FLAGS       "-mp -mp=bind,allcores,numa" )
+# Note: OpenMP_<lang>_FLAGS gets overwritten by the FindOpenMP module
+# unless its stored as a cache variable
+set( OpenMP_C_FLAGS             "-mp -mp=bind,allcores,numa" CACHE STRING "" FORCE )
+set( OpenMP_CXX_FLAGS           "-mp -mp=bind,allcores,numa" CACHE STRING "" FORCE )
+set( OpenMP_Fortran_FLAGS       "-mp -mp=gpu,bind,allcores,numa" CACHE STRING "" FORCE )
```

Comment:
What if you remove `FORCE`?
If that works, then that would give you option to override via command-line still.

Reply:
Inded, `FORCE` is not required here. However, the situation is even worse and inconsistent to specify custom OpenMP flags:

* For Fortran, the easiest way is to set the variable in the cache, thus keeping the module from overwriting it
* For C/CXX, it requires setting `OpenMP_${LANG}_FLAGS` _and_ `OpenMP_${LANG}_LIB_NAMES` to shortcut the derivation of these two variables entirely, otherwise they get overwritten with `FORCE` by the module.

The latter feels very unnecessary and I might open an issue about it with Kitware.

Reply:
For demonstration purposes I'll put it in the arch files here with a comment to refer to later on, as the ability to overwrite them may be crucial to certain offloading configurations, where the auto-discovery of flags may just be wrong.

Culprit is [here](https://github.com/Kitware/CMake/blob/c4cdd7a24d3016ccf91fae26646a548b52de4a78/Modules/FindOpenMP.cmake#L475) in the module

Reply:
right, thanks for the reminder!

Reply:
Tracked in https://gitlab.kitware.com/cmake/cmake/-/issues/23743